plugins/modules/collaborator_information.py

```diff
@@ -0,0 +1,352 @@
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.type:
+    description: This will return what type of collaborator the user is.
+    type: str
+    returned: only if at least one collaborator is contained within repository
+'''
+
+
+__metaclass__ = type
+
+
+def present_collaborator(g, repo, collaborator, permission):
+    r = g.get_repo(repo)
+    r.add_to_collaborators(collaborator, permission=permission)
+
+
+def absent_collaborator(g, repo, collaborator):
+    r = g.get_repo(repo)
+    r.remove_from_collaborators(collaborator)
+
+
+def get_collaborators(g, repo):
+    dict_repo = list()
+    collab_output = dict()
+    collaborators = g.get_repo(
+        repo).get_collaborators(affiliation="direct")
+    for collaborator in collaborators:
+        collab_output['login'] = collaborator.login
+        collab_output['id'] = collaborator.id
+        collab_output['type'] = collaborator.type
+        collab_output['site_admin'] = collaborator.site_admin
+        permissions = {
+            'triage': collaborator.permissions.triage,
+            'push': collaborator.permissions.push,
+            'pull': collaborator.permissions.pull,
+            'admin': collaborator.permissions.admin
+        }
+        collab_output['permissions'] = permissions
+
+        dict_repo.append(collab_output.copy())
+
+    return dict_repo
+
+
+def present_collaborator_check_mode(g, repo, collaborator, permission, current_collaborators):
+    collaborator_position = next((i for i, x in enumerate(
+        current_collaborators) if x["login"] == collaborator), None)
+    permissions = {}
+    if permission == 'admin':
+        permissions = {
+            'admin': True,
+            'pull': True,
+            'push': True,
+            'triage': True
+        }
+    elif permission == 'pull':
+        permissions = {
+            'admin': False,
+            'pull': True,
+            'push': False,
+            'triage': False
+        }
+    else:
+        permissions = {
+            'admin': False,
+            'pull': True,
+            'push': True,
+            'triage': True
+        }
+
+    output_collaborators = current_collaborators.copy()
+    if collaborator_position is None:
+        # adding
+        collaborator_to_add = {
+            'login': collaborator,
+            'id': 000,
+            'type': 'User',
+            'site_admin': True if permission == 'admin' else False,
+            'permissions': permissions
+        }
+        output_collaborators.append(collaborator_to_add)
+
+    else:
+        # changing
+        for current_collaborator in output_collaborators:
+            if collaborator == current_collaborator['login']:
+                current_collaborator['permissions'] = permissions
+                if permission == 'admin':
+                    current_collaborator['site_admin'] = True
```

Comment:
Can the collaborator_position be used here like it is in the absent_collaborator_check_mode?  I'm trying to understand why it was necessary to iterate over output_collaborators?